### PR TITLE
Basic documentation for RTD.

### LINKED ISSUE #10: Publish docs
Apparently I got so far as setting up the RTD site but never actually enabled the sphinx configuration.

### PR DESCRIPTION
**Pull request by jamadden**
Using the existing documentation files and docstrings.

Fixes #10. 

Viewable at http://zopeappapidoc.readthedocs.io/en/issue10/

### CODE REVIEW

File: src/zope/app/apidoc/codemodule/browser/introspector.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ class TraversalRoot(object):
 
 
 class Introspector(BrowserView):
+    "Introspector browser view"
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/introspector.py]: """ ... """ ?

Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/introspector.py]: (Doesn't pylint catch this?  I'm 80% sure flake8 would.)

File: src/zope/app/apidoc/codemodule/browser/introspector.rst [hunk 2]
```diff
@@ -27,20 +27,22 @@ and the name of the object:
 Of course, the root folder does not have a name. As you can see the type links
 directly to the API documentation of the class.
 
-The next section lists all directly provided interfaces. The root folder
-directly provides the ``ISite`` and ``IRootFolder`` interface, so we should
-see those:
+The next section lists all directly provided interfaces. The root
+folder directly provides the :class:`zope.site.interfaces.ISite` and
+:class:`zope.site.interfaces.IRootFolder` interface, so we should see
+those::
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/browser/introspector.rst]: So now I'm wondering what's the difference in Sphinx rendering regular doctest blocks (ReStructuredText [supports these natively](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#doctest-blocks)) and literal blocks that happen to contain doctests.

In other words, why are you adding `::` before each doctest block?

File: src/zope/app/apidoc/codemodule/codemodule.py [hunk 3]
```diff
@@ -58,8 +58,9 @@ class is found, it is represented as a bold entry in the list.
     interface that requires a method or attribute to be implemented and the
     permissions required to access it.
     """)
+
     def __init__(self):
-        """Initialize object."""
+        "Initialize object."
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/codemodule.py]: This doesn't seem right!

File: src/zope/app/apidoc/codemodule/directives.rst [hunk 4]
```diff
@@ -37,13 +37,13 @@ and the root module is available:
...
+The ``apidoc:importModule`` Directive
+=====================================
 
-The `importModule` directive allows you to set the
+The ``importModule`` directive allows you to set the
 ``__import_unknown_modules__`` flag of the class registry. When this flag is
-set to false, paths will only be looked up in ``sys.modules``. When set true,
-and the ``sus.modules`` lookup fails, the import function of the class
+set to false, paths will only be looked up in :data:`sys.modules`. When set true,
+and the :data:`sus.modules` lookup fails, the import function of the class
```
Comment (mgedmin) [inline src/zope/app/apidoc/codemodule/directives.rst]: The `sus.modules` looks like a typo.

File: src/zope/app/apidoc/utilities.py [hunk 5]
```diff
@@ -48,6 +48,7 @@
 
 
 def relativizePath(path):
+    "Convert the path to a relative form."
```
Comment (mgedmin) [inline src/zope/app/apidoc/utilities.py]: """" ... """?

File: src/zope/app/apidoc/utilitymodule/utilitymodule.py [hunk 6]
```diff
@@ -34,12 +34,14 @@
 NONAME = '__noname__'
 
 def encodeName(name):
+    "Encode the name in URL safe base 64."
```
Comment (mgedmin) [inline src/zope/app/apidoc/utilitymodule/utilitymodule.py]: """ ... """?

File: src/zope/app/apidoc/utilitymodule/utilitymodule.py [hunk 7]
```diff
@@ -34,12 +34,14 @@
...
     name = base64.urlsafe_b64encode(name.encode('utf-8'))
     if not isinstance(name, str):
         name = name.decode('ascii')
     return name
 
 def decodeName(name):
+    "Decode the name as encoded by :func:`encodeName`."
```
Comment (mgedmin) [inline src/zope/app/apidoc/utilitymodule/utilitymodule.py]: """ ... """?

File: README.rst [hunk 8]
```diff
@@ -1,7 +1,17 @@
+.. image:: https://readthedocs.org/projects/zopeappapidoc/badge/?version=latest
+        :target: http://zopeappapidoc.readthedocs.io/en/latest/?badge=latest
```
Comment (jamadden) [inline README.rst]: `http` is what you get if you click the badge icon on RTD and copy-and-paste, which is what I did here. Presumably I then copy-and-pasted into CHANGES too.

I don't know if this means RTD prefers http (less overhead on the infrastructure?) or if they don't care. Anyway, I'll change it, thanks for the catch.

untitled

File: src/zope/app/apidoc/codemodule/browser/introspector.py (hunk 1 above)
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/browser/introspector.py]: pylint does not have a warning for that. Neither does flake8 as far as I can see:
```
$ flake8 --isolated src/zope/app/apidoc/codemodule/browser/introspector.py
src/zope/app/apidoc/codemodule/browser/introspector.py:33:1: E302 expected 2 blank lines, found 1
src/zope/app/apidoc/codemodule/browser/introspector.py:34:14: E721 do not compare types, use 'isinstance()'
src/zope/app/apidoc/codemodule/browser/introspector.py:111:49: E225 missing whitespace around operator
```

Still, PEP 257 [does say](https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring):

> For consistency, always use `"""triple double quotes"""` around docstrings.

I'm horribly inconsistent in my use of `"`, `'` and apparently `"""`, just doing whatever is easiest at the time. I'll try to be more consistent with one-line docstrings. Thanks for pointing it out.

PS: Here's pylint's default output for that same file. I have a number of those warnings disabled by default when I'm working in the zope code base:
```
$ pylint -r no src/zope/app/apidoc/codemodule/browser/introspector.py
************* Module zope.app.apidoc.codemodule.browser.introspector
C:158, 0: Wrong hanging indentation (remove 13 spaces).
                                 name, klass._Class__all_ifaces)
                    |            ^ (bad-continuation)
C:184, 0: Wrong hanging indentation (remove 1 space).
                     val.__doc__ or '',
                    |^ (bad-continuation)
C:185, 0: Wrong hanging indentation (remove 1 space).
                     getParent(self.klassView.context).getPath()),
                    |^ (bad-continuation)
C:187, 0: Wrong hanging indentation (remove 1 space).
                     name, klass._Class__all_ifaces)}
                    |^ (bad-continuation)
C: 33, 0: Invalid function name "getTypeLink" (invalid-name)
C: 33, 0: Missing function docstring (missing-docstring)
R: 38, 4: Consider using ternary (path.replace('.', '/') if importable else None) (consider-using-ternary)
C: 41, 0: Invalid class name "annotationsNamespace" (invalid-name)
W: 44,27: Unused argument 'request' (unused-argument)
C: 47, 4: Missing method docstring (missing-docstring)
W: 47,29: Unused argument 'ignore' (unused-argument)
R: 41, 0: Too few public methods (1/2) (too-few-public-methods)
C: 59, 0: Invalid class name "sequenceItemsNamespace" (invalid-name)
W: 62,27: Unused argument 'request' (unused-argument)
C: 65, 4: Missing method docstring (missing-docstring)
W: 65,29: Unused argument 'ignore' (unused-argument)
R: 59, 0: Too few public methods (1/2) (too-few-public-methods)
C: 72, 0: Invalid class name "mappingItemsNamespace" (invalid-name)
W: 81,27: Unused argument 'request' (unused-argument)
C: 84, 4: Missing method docstring (missing-docstring)
W: 84,29: Unused argument 'ignore' (unused-argument)
R: 72, 0: Too few public methods (1/2) (too-few-public-methods)
C: 92, 0: Missing class docstring (missing-docstring)
R: 92, 0: Too few public methods (0/2) (too-few-public-methods)
C:109, 8: Invalid attribute name "klassView" (invalid-name)
C:115, 4: Missing method docstring (missing-docstring)
C:118, 4: Invalid method name "getBaseURL" (invalid-name)
C:118, 4: Missing method docstring (missing-docstring)
C:121, 4: Invalid method name "getDirectlyProvidedInterfaces" (invalid-name)
C:121, 4: Missing method docstring (missing-docstring)
C:127, 4: Invalid method name "getProvidedInterfaces" (invalid-name)
C:127, 4: Missing method docstring (missing-docstring)
C:130, 4: Invalid method name "getBases" (invalid-name)
C:130, 4: Missing method docstring (missing-docstring)
C:133, 4: Invalid method name "getAttributes" (invalid-name)
C:133, 4: Missing method docstring (missing-docstring)
W:158,39: Access to a protected member _Class__all_ifaces of a client class (protected-access)
C:164, 4: Invalid method name "getMethods" (invalid-name)
C:164, 4: Missing method docstring (missing-docstring)
W:187,27: Access to a protected member _Class__all_ifaces of a client class (protected-access)
C:194, 4: Invalid method name "isSequence" (invalid-name)
C:194, 4: Missing method docstring (missing-docstring)
C:198, 4: Invalid method name "getSequenceItems" (invalid-name)
C:198, 4: Missing method docstring (missing-docstring)
C:212, 4: Invalid method name "isMapping" (invalid-name)
C:212, 4: Missing method docstring (missing-docstring)
C:216, 4: Invalid method name "getMappingItems" (invalid-name)
C:216, 4: Missing method docstring (missing-docstring)
C:230, 4: Invalid method name "isAnnotatable" (invalid-name)
C:230, 4: Missing method docstring (missing-docstring)
C:233, 4: Invalid method name "getAnnotationsInfo" (invalid-name)
C:233, 4: Missing method docstring (missing-docstring)
E:239,26: Too many positional arguments for method call (too-many-function-args)
C: 20, 0: standard import "import inspect" should be placed before "import six" (wrong-import-order)

-----------------------------------
Your code has been rated at 4.77/10
```

File: src/zope/app/apidoc/codemodule/directives.rst (hunk 4 above)
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/directives.rst]: Hehe. The squiggle underline of a misspelled word made `sys` and `sus` look identical in my font.

File: src/zope/app/apidoc/codemodule/browser/introspector.rst (hunk 2 above)
Comment (jamadden) [inline src/zope/app/apidoc/codemodule/browser/introspector.rst]: Whitespace in the block in rendered differently by Sphinx. With one colon, you get a separate section every time there's a blank line:
one

But with two colons, the entire block is treated as one example when rendered:
two

I think the two-colon version is less cluttered looking and also expresses the intent that this is meant to be read as one example better. The [sphinx docs](http://www.sphinx-doc.org/en/stable/rest.html#source-code) seem to suggest that `::` is the preferred spelling.

Plus, Emacs highlights the two types of blocks differently in the source 😄

**Comment by jamadden** [comment]
Thanks for the feedback! I've addressed it and will merge.

**Pull request merged**